tutorials/s4sdk-caching/s4sdk-caching.md

```diff
@@ -51,268 +56,136 @@ If each of these options apply to your use case, then we highly recommend the ca
+1. Determine how long objects are to be cached
+2. Declare the parameters that need to be stored together with the cached data
 
-import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartner;
+The first step is obviously necessary, since we want to set up a cache that is meant to store data only a short amount of time to optimize execution time. How long you want to keep data cached depends on the data that you receive. How fast do you expect the information to be outdated? How frequently will the data be accessed?
 
-@WebServlet("/businesspartners")
-public class BusinessPartnerServlet extends HttpServlet {
+Secondly, we specify the parameters that are to be cached with the data. For our request to retrieve a list of business partners no parameters are necessary, so we build our cache `wihtoutParameters`. But imagine you want to fetch information about a specific business partner by passing an ID to the system. In order to cache such a request the cache needs to not only remember the result received, but also the ID that was associated with it. In such a case one can simply pass such parameters by using `.withParameters(param1, param2, ..)`.
 
-    private static final long serialVersionUID = 1L;
-    private static final Logger logger = CloudLoggerFactory.getLogger(BusinessPartnerServlet.class);
+Feel free to test that subsequent requests respond faster compared to the first request issued. Deploy your application locally or in the cloud and access the list of business partners a couple of times.
 
-    @Override
-    protected void doGet(final HttpServletRequest request, final HttpServletResponse response)
-            throws ServletException, IOException {
-        try {
-            final List<BusinessPartner> businessPartners =
-                    new GetCachedBusinessPartnersCommand().execute();
-
-            response.setContentType("application/json");
-            response.getWriter().write(new Gson().toJson(businessPartners));
+[DONE]
+[ACCORDION-END]
 
-        } catch (final Exception e) {
-            logger.error(e.getMessage(), e);
-            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-            response.getWriter().write(e.getMessage());
-        }
-    }
+[ACCORDION-BEGIN [Step 5: ](Test the Cache)]
+
+Now that we have a working command with caching functionality we also have to adapt our test. Recall the test we prepared to check our resilient command falls back to an empty list in case of failure. Note that this behavior has now changed slightly.
+
+If our servlet got the desired result cached from a previous call, and the ERP system is temporarily not available, our cache will still return the data. But the test expects the result to be empty in that case. In order to account for this behavior and to see if our cache is working as expected let's adapt the test to account for caching. Replace the `testWithFallback` test with the following code:
```

Comment:
Why replace the test instead of adding a new one and adapt the existing one (in the next section)? The fallback functionality still needs to be tested.